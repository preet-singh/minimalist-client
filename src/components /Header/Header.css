.Header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: rgba(255,255,255,0.1);
    margin: 0px;
    padding: 0px;
    margin-bottom: 30px;
}

.Header a { 
    width: 100%;
}

.header-o {
    height: 35px;
    width: 35px;
    align-self: flex-end;
    margin-left: 2px;
    text-shadow: 0px 6px 8px #005415;
}
h1 {
    margin: 0px;
    padding: 0px;
    color: #00d938;
    font-size: 60px;
    line-height: 1;
    padding: 10px;
    position: relative;
    display: block;
    width: 100%;
    text-shadow: 0px 6px 8px #005415;
    font-family: 'Fugaz One', cursive;
}

.Header a { 
    color: white;
    text-decoration: none;
}

.Header h1 span { 
    display: block;
    display: flex;
    align-content: center;
    justify-content: center;
    font-size: 34px;
    font-family: 'Fugaz One', cursive;
}

.Header h1:hover, .Header h1:hover span {
    color: white;
}

.Header .about-link {
    font-weight: bold;
    margin-right: 15px;
    display: inline;
}

.Header .log-button {
    margin-left: 15px;
}

.under-bar {
    display: flex;
}

@media screen and (min-width:768px){
    .Header {
        background: linear-gradient(to right, rgba(255,255,255,0),rgba(255,255,255,0.1),rgba(255,255,255,0));
    }

    h1 {
        font-size: 80px;
    }

}