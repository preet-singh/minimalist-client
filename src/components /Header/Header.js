import React, { Component } from 'react'
import { Link, withRouter } from 'react-router-dom'
import { Hyph } from '../../utils/utils'
import TokenService from '../../services/token-service'
import IdleService from '../../services/idle-service'
import LandingPage from '../LandingPage/LandingPage';
import InventoryContextNew from '../../InventoryContextNew';
//import './Header.css'

class Header extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    TokenService.clearCallbackBeforeExpiry()
    IdleService.unRegisterIdleResets()
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link
          to='/register'>
          Register
        </Link>
        <Hyph />
        <Link
          to='/login'>
          Log in
        </Link>
      </div>
    )
  }

  // render() {
  //   return (
  //     <nav className='Header'>
  //       <h1>
  //         <Link to='/'>
  //           Minimalist
  //         </Link>
  //       </h1>
  //       {TokenService.hasAuthToken()
  //         ? this.renderLogoutLink()
  //         : this.renderLoginLink()}
  //     </nav>
  //   )
  // }
  static contextType = InventoryContextNew;
  render() {
  return(
    <nav className="Header">
      <h1><Link to=''>Minimalist</Link></h1>
         {TokenService.hasAuthToken()
          ? this.renderLogoutLink()
          : this.renderLoginLink()}
      <div className="under-bar"><Link to='/' onClick={(e) => {
        window.localStorage.removeItem('minimalistSeenLanding');
        this.context.setLanding(false);
        }} className="about-link">About</Link>
      </div>
    </nav>
  );
    }
}

export default withRouter(Header)