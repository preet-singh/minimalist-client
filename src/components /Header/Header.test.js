import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Header from './Header'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Add inventory testing', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>,
        div)
    ReactDOM.unmountComponentAtNode(div)
    });

    it('renders the user inventory', () => {
        const wrapper = shallow(<Header  />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
});