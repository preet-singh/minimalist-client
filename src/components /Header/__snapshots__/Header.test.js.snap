// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add inventory testing renders the user inventory 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
